### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by michalsidzej**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+language: node_js
```
Comment (kowalski) [inline .travis.yml]: not sure why we need this file?
we are using github actions not travis

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+# Buidler to Hardhat plugin migration guide
```
Comment (kowalski) [inline MIGRATION.md]: I'm not sure if this file is relevant?

File: README-TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+# hardhat-example-plugin
```
Comment (kowalski) [inline README-TEMPLATE.md]: and this?

File: README.md [hunk 4]
```diff
@@ -1 +1,92 @@
...
+- `ethAccountLockCodeHash`,
+- `privateKey`
+
+This is an example of how to set it in `hardhat.config.ts`:
+
+```ts
+export const config: HardhatUserConfig = {
+  network: {
+    godwoken: {
+      url: "http://localhost:8024",
```
Comment (kowalski) [inline README.md]: remember to update example config after you move attributes specific to godwoken under godwoken tree

File: src/deploy.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          options.args,
+          contractArtifact.abi,
+          contractArtifact.bytecode
+        );
+      }
+
+      return oldDeploy(name, options);
+    };
+    hre.deployments.deploy = newDeploy;
+  } else {
```
Comment (kowalski) [inline src/deploy.ts]: it's more readable when you revert it.
you have:

```
if (importantCondition) {
  doStuff()
} else {
  reportError()
}
```

It's definitely recommended to always to this as:
```
if (!importantCondition) {
   reportError()
   return
}
doStuff()
```

File: src/type-extensions.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import "hardhat/types/config";
+
+declare module "hardhat/types/config" {
+  export interface HttpNetworkUserConfig {
+    privateKey?: string;
```
Comment (kowalski) [inline src/type-extensions.ts]: define sth like `GodwokenUserConfig` and have here:
```
  godwoken: GodwokenUserConfig
```

Comment (michalsidzej) [reply src/type-extensions.ts:5]: Zapiszę to jako polyjuiceConfig

**Review by kowalski — APPROVED** [review]

**Pull request merged**